pkg/volume/local_volume/local_volume.go

```diff
@@ -0,0 +1,273 @@
+	return []v1.PersistentVolumeAccessMode{
+		v1.ReadWriteOnce,
+	}
+}
+
+func getVolumeSource(spec *volume.Spec) (*v1.LocalVolumeSource, bool, error) {
+	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.LocalVolume != nil {
+		return spec.PersistentVolume.Spec.LocalVolume, spec.ReadOnly, nil
+	}
+
+	return nil, false, fmt.Errorf("Spec does not reference an LocalVolume volume type")
+}
+
+func (plugin *localVolumePlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+	volumeSource, readOnly, err := getVolumeSource(spec)
+	if err != nil {
+		return nil, err
+	}
+
+	if volumeSource.Fs == nil {
+		return nil, fmt.Errorf("LocalVolume.Fs is nil")
+	}
+
+	hostNode := plugin.host.GetNodeName()
+	if volumeSource.NodeName != hostNode {
+		return nil, fmt.Errorf("LocalVolume NodeName %q does not match this node %q", volumeSource.NodeName, hostNode)
+	}
+
+	return &localVolumeMounter{
+		localVolume: &localVolume{
+			podUID:     pod.UID,
+			volName:    spec.Name(),
+			mounter:    plugin.host.GetMounter(),
+			plugin:     plugin,
+			globalPath: volumeSource.Fs.Path,
+		},
+		readOnly: readOnly,
+	}, nil
+
+}
+
+func (plugin *localVolumePlugin) NewUnmounter(volName string, podUID types.UID) (volume.Unmounter, error) {
+	return &localVolumeUnmounter{
+		localVolume: &localVolume{
+			podUID:  podUID,
+			volName: volName,
+			mounter: plugin.host.GetMounter(),
+			plugin:  plugin,
+		},
+	}, nil
+}
+
+func (plugin *localVolumePlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
+	localVolume := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: volumeName,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				LocalVolume: &v1.LocalVolumeSource{
+					NodeName: plugin.host.GetNodeName(),
+					Fs:       &v1.LocalFsVolume{Path: ""},
+				},
+			},
+		},
+	}
+	return volume.NewSpecFromPersistentVolume(localVolume, false), nil
+}
+
+// Local volumes represent a local directory on a node.
+// The directory at the globalPath will be bind-mounted to the pod's directory
+type localVolume struct {
+	volName string
+	podUID  types.UID
+	// Global path to the volume
+	globalPath string
+	// Mounter interface that provides system calls to mount the global path to the pod local path.
+	mounter mount.Interface
+	plugin  *localVolumePlugin
+	// TODO: add metrics
+	volume.MetricsNil
+}
+
+func (l *localVolume) GetPath() string {
+	return l.plugin.host.GetPodVolumeDir(l.podUID, strings.EscapeQualifiedNameForDisk(localVolumePluginName), l.volName)
+}
+
+type localVolumeMounter struct {
+	*localVolume
+	readOnly bool
+}
+
+var _ volume.Mounter = &localVolumeMounter{}
+
+func (m *localVolumeMounter) GetAttributes() volume.Attributes {
+	return volume.Attributes{
+		ReadOnly:        m.readOnly,
+		Managed:         false,
+		SupportsSELinux: false,
```

Comment:
In SELinux enabled cluster, we should have this set to true in order to share local volumes even for containers in a pod.  This is also required to share volume among different pods.

We already have user manage and provide SELinux labels right now, so setting this to true should suffice.

also commented here https://github.com/kubernetes/kubernetes/issues/45054